client/src/templates/Challenges/utils/frame.ts

```diff
@@ -273,11 +273,47 @@ const initTestFrame = (frameReady?: () => void) => (frameContext: Context) => {
 const initMainFrame =
   (_: unknown, proxyLogger?: ProxyLogger) => (frameContext: Context) => {
     waitForFrame(frameContext)
-      .then(() => {
+      .then(async () => {
+        const iframeBuild: string = frameContext.build;
+        const parser = new DOMParser();
+        const iframeBuildDoc = parser.parseFromString(iframeBuild, 'text/html');
+        const allLinks = iframeBuildDoc.querySelectorAll(
+          'script[src], link[href]'
+        );
+        let errMsgs = '';
+        let unfetchedLinks = allLinks.length;
+
+        if (!unfetchedLinks) return null;
+
+        for (const url of allLinks) {
+          const urlAddr =
+            url.tagName === 'SCRIPT'
+              ? (url as HTMLScriptElement).src
+              : (url as HTMLLinkElement).href;
+
+          try {
+            const response = await fetch(String(urlAddr));
+            if (!response.ok) {
+              errMsgs = `${errMsgs}\nCannot retrieve ${urlAddr}, link status code: ${response.status}`;
+            } else if (!unfetchedLinks) {
+              continue;
+            }
+            unfetchedLinks -= 1;
```

Comment:
```suggestion
            return response.ok
              ? ''
              : `Cannot retrieve ${url}, link status code: ${response.status}`;
```
What was the logic behind updating `unfetchedLinks`? It seems to me that either all the links can be fetched (and we're fine) or not (and we should show an error). I couldn't see how `unfetchedLinks` fit into that.

Reply:
The idea behind `unfetchedLinks` is that I want to gather all error messages from multiple fetching errors before displaying it in the `proxyLogger`. But I think it should be ok to get rid of it.